### PR TITLE
feat: add versioned documentation

### LINKED ISSUE #432: Add versioning of documentation
The documentation of calcs and objects should be connected to a specific version of Septic. This would make it possible for the user to select which version of Septic they are using and get the relevant diagnostics for that version. Might be useful when upgrading a config to a new version of Septic.

### PR DESCRIPTION
**Pull request by vetlek**
Closes #432

### CODE REVIEW

**Comment by vetlek** [comment]
Might require some updates to the automatic documentation update action.

File: scripts/main.py [hunk 1]
```diff
@@ -1,52 +1,151 @@
...
+    package["contributes"]["configuration"]["properties"][
+        "septic.documentation.version"
+    ]["enum"] = list(map(folder_name_to_option, get_versions(output_path)))
+    package_path = Path("package.json")
+    with open(package_path.resolve(), "w") as f:
+        json.dump(package, f, indent=2)
 
 
-def updateObjects():
+def updateObjects(ref: str, output_path: Path):
```
Comment (einarsi) [inline scripts/main.py:107]: Is `ref` a commit hash or a github ref? It looks like it is the former, in which case `commit` may be a better variable name for readability? Same for `updateCalcs()`

File: scripts/src/github.py [hunk 2]
```diff
@@ -32,41 +32,47 @@ def getCommitId(branch: str):
     repo = "SEPTIC"
     endpoint = f"/repos/{owner}/{repo}/branches/{branch}"
     response = sendRequestGithub(endpoint)
-    return response["commit"]["sha"][0:7]
+    return response["commit"]["sha"]
 
 
-def getCalcFile(branch: str):
+def getCalcFile(ref: str):
```
Comment (einarsi) [inline scripts/src/github.py:38]: `ref` or `commit` as mentioned above? Same for `getFile()`, `getDir()`, `getObjectFiles()`,

File: scripts/src/versioning.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+def folder_name_to_option(name: str):
+    return name.replace("_", ".")
+
+
+def version_to_folder_name(version: tuple):
+    major = get_major(version)
+    return "v" + "_".join([str(x) for x in major])
+
+
+def version_to_setting_name(version: tuple):
```
Comment (einarsi) [inline scripts/src/versioning.py:20]: Wondered what this does based on the function name, but looks like it's not in use? If so: Remove?

File: server/src/septic/septicMetaInfo.ts [hunk 4]
```diff
@@ -148,13 +165,12 @@ export class SepticMetaInfoProvider {
...
-        string,
-        ISepticObjectDocumentation
-    > {
+    private loadObjectsDocumentation(
+        version: string
+    ): Map<string, ISepticObjectDocumentation> {
         const filePath = path.join(
             __dirname,
-            "../../../public/objectsDoc.yaml"
+            `../../../public/${version.replace(/\./g, "_")}/objectsDoc.yaml`
```
Comment (einarsi) [inline server/src/septic/septicMetaInfo.ts:173]: Out of interest: Does the join() method properly handle joining paths with "../"? I.e. does join("/home/user", "../../root") turn into "/root" or "/home/user/../../root"?

**Review by einarsi — CHANGES_REQUESTED** [review]

File: package.json [hunk 5]
```diff
@@ -38,6 +38,16 @@
           "default": "off",
           "description": "Traces the communication between VS Code and the language server."
         },
+        "septic.documentation.version": {
+          "scope": "window",
+          "type": "string",
+          "enum": [
+            "latest",
+            "v2.88"
```
[the remark this answers is not in the document]
Comment (vetlek) [reply package.json:46]: Updated automatically

File: scripts/main.py (hunk 1 above)
Comment (vetlek) [reply scripts/main.py:107]: It is a github ref. I think both commit and branch is used in the scripts

File: server/src/septic/septicMetaInfo.ts (hunk 4 above)
Comment (vetlek) [reply server/src/septic/septicMetaInfo.ts:173]: Yes, I think that pathlib handles "../" correctly

File: scripts/src/versioning.py (hunk 3 above)
Comment (vetlek) [reply scripts/src/versioning.py]: Not used, removed

**Review by einarsi — APPROVED** [review]
💯

**Pull request merged**